.github/workflows/hlint.yaml

```diff
@@ -0,0 +1,22 @@
+name: hlint
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  hlint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: 'Set up HLint'
+      uses: haskell/actions/hlint-setup@v2
+      with:
```

Comment:
Previously we failed the check run whenever hlint reported anything, so we should keep that unless that's a deliberate change in behaviour.

```
fail-on: status
```

Reply:
It was deliberate, actually, since I think some of hlint's suggestions are a bit too opinionated, but maybe you're right and I should just suffer through and fix the new ones :)

Reply:
Now that we're using the correct hlint version, there are no suggestions, and I suspect you could even drop 2c95e27 if you wanted to and there still wouldn't be any. That doesn't change the fact hlint could be opinionated in general, but it no longer produces suggestions that we'd rather not implement…

Reply:
Interesting that we'd get _less_ warnings with a newer hlint version (considering that the suggestions I saw were valid, just perhaps a bit overambitious)

Reply:
Yeah, old me would go and find out what's going on… :-)